src/researches/getKeywordDensity.js

```diff
@@ -1,19 +1,29 @@
 /** @module analyses/getKeywordDensity */
 
 const countWords = require( "../stringProcessing/countWords.js" );
+const keyphraseLength = require( "./keyphraseLength" );
 
 /**
- * Calculates the keyword density .
+ * Calculates the keyword density.
  *
  * @param {object} paper The paper containing keyword and text.
  * @param {object} researcher The researcher.
  * @returns {number} The keyword density.
  */
 module.exports = function( paper, researcher ) {
 	const wordCount = countWords( paper.getText() );
+	const lengthKeyphrase = keyphraseLength( paper );
+
 	if ( wordCount === 0 ) {
 		return 0;
 	}
 	const keywordCount = researcher.getResearch( "keywordCount" );
-	return ( keywordCount / wordCount ) * 100;
+
+	/*
+	 * This formula consists of the basic keyword density formula
+	 * (keywordCount / wordCount) * 100, with an added factor for
+	 * keyphrase length. Keyphrase length gets a base weight of 0.7, plus a
+	 * specific weight for the keyphrase length (lengthKeyphrase / 3).
+	 */
+	return ( keywordCount / wordCount ) * 100 * ( 0.7 + lengthKeyphrase / 3 );
```

Comment:
I've seen a part of this formula [before](https://github.com/Yoast/YoastSEO.js/pull/1411/files#diff-1fe2b1c3965158b2d0f682d12d347131R18): `100 * ( 0.7 + lengthKeyphrase / 3 )`. You might consider making extracting this.